charts/artifact-caching-proxy/templates/nginx-proxy-configmap.yaml

```
@@ -6,30 +6,85 @@ metadata:
 data:
   cache.conf: |
     proxy_cache_path {{ .Values.cache.path }} levels=1 keys_zone=nginx_cache:{{ .Values.cache.keysZoneSize }} max_size={{ .Values.persistence.size }}g inactive={{ .Values.cache.inactive }} use_temp_path={{ .Values.cache.useTempPath }};
-  proxy.conf: |
-    ## add ssl entries when https has been set in config
-    ## server configuration
+  common-proxy.conf: |
+    # Enable caching
+    proxy_cache         nginx_cache;
+
+    # Specify HTTP code of responses to cache
+    proxy_cache_valid {{ .Values.proxy.proxyCacheValidCode }} {{ .Values.proxy.proxyCacheValidCodeDuration }};
+
+    # Ensure that no authentication header are sent to the upstreams
+    proxy_set_header    Authorization     "";
+
+    # Pass headers to allow HTTP/30x responses from upstream to be followed
+    proxy_set_header    X-Artifactory-Override-Base-Url $http_x_forwarded_proto://$host:$server_port/;
+    proxy_set_header    X-Forwarded-Port  $server_port;
+    proxy_set_header    X-Forwarded-Proto $http_x_forwarded_proto;
+    proxy_set_header    X-Forwarded-For   $proxy_add_x_forwarded_for;
+
+    # Pass the current Server's identification to upstream
+    # Ref. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server
+    proxy_pass_header   Server;
+
+    # Remove any cookie in the requests to upstream
+    proxy_cookie_path   ~*^/.* /;
+
+    proxy_read_timeout  900;
+  vhost-proxy.conf: |
+    include /etc/nginx/conf.d/cache.conf;
 
     server {
         listen {{ .Values.service.port }} default;
+        server_name _; # Catch all hostnames, including invalids
+
+        # Specify a DNS resolver ton ensure upstream DNS are dynamically resolved
+        # Ref. https://github.com/DmitryFillo/nginx-proxy-pitfalls
+        resolver 8.8.8.8 valid=60s;
```

shouldn't we use 
```suggestion
        resolver 1.1.1.1 valid=60s;
```

---

Google or CloudFlare, pick your poison ^^

Doesn't really matter IMO

---

```suggestion
        resolver 9.9.9.9 valid=60s;
```

Good catch!

Switchging to Quad 9: https://www.quad9.net/